spark/src/main/scala/geotrellis/spark/io/hadoop/HadoopLayerProvider.scala

```diff
@@ -59,4 +59,13 @@ class HadoopLayerProvider extends AttributeStoreProvider
     val interval = params.getOrElse("interval", "4").toInt
     new HadoopLayerWriter(path, store, interval)
   }
+
+  def valueReader(uri: URI, store: AttributeStore): ValueReader[LayerId] = {
+    val _uri = trim(uri)
+    val path = new Path(_uri)
+    val params = UriUtils.getParams(_uri)
```

Comment:
For `valueReader`, `layerWriter`, `attribuetStore` 

```scala
val _uri = trim(uri)
val path = new Path(_uri)
val params = UriUtils.getParams(_uri)
```

Mb that can be sugared?